React Revamp: React Introduction (#24636)
TheOdinProject/curriculum: react/react_introduction_rewrite.md

### Introduction

This lesson will provide you with information on what React is, its brief history, and why we're choosing to cover it in our curriculum.

### Lesson Overview

> This section contains a general overview of topics that you will learn in this lesson.
> 
> - Understand the differences between a Javascript Library and Framework
> - Understand the benefits of learning React
> - Understand what is react

Reviewer: We can clean this up. I believe we really only require one Lesson Overview:
```suggestion
This section contains a general overview of topics that you will learn in this lesson.

- Understand what is React
```

### What Is React?

> React according to their <a href="https://reactjs.org/"> official website </a> is "A JavaScript library for building user interfaces" however lets break this down a bit further.

Reviewer: Need the markdown formatted link here.
```suggestion
React according to their <a href="https://reactjs.org/"> official website </a> is "A JavaScript library for building user interfaces" however lets break this down a bit further.
```

Generally speaking, A JavaScript library is a collection of code that's designed to make our lives easier, not to be confused with frameworks however as you'll be reading the differences later on in the lesson. A JavaScript Library contains pre-written code "snippets" that can be reused/reimplemented in our own ways to make building things a little bit easier for ourselves.

Since we know that ReactJS is used for building user interfaces. We can come to the conclusion that ReactJS is a JavaScript Library in which their pre-written code base help us build user interfaces easier and quicker.

### Why Cover React?

React is one of the most powerful, widely used frontend frameworks.

The landscape for frontend frameworks has been changing a lot over the last few years, so it is understandable to be worried about choosing the "wrong" one.

> [This article](https://stackoverflow.blog/2018/01/11/brutal-lifecycle-javascript-frameworks/) shows the recent development of frontend frameworks well. Once you start diving deeper into a framework, you will begin to love it. It makes your code easily scalable, more readable, and possibly a thousand times more efficient (just our modest estimation).

Reviewer: Link text suggestion:
```suggestion
[This article on the lifecycle of JavaScript frameworks](https://stackoverflow.blog/2018/01/11/brutal-lifecycle-javascript-frameworks/) shows the recent development of frontend frameworks well. Once you start diving deeper into a framework, you will begin to love it. It makes your code easily scalable, more readable, and possibly a thousand times more efficient (just our modest estimation).
```

> Just to name a few reasons on why to learn React.js:

Reviewer: I think we should stick to "React" rather than "ReactJS" or "React.js"

```suggestion
Just to name a few reasons on why to learn React:
```

- Reusability of components
- Well supported due to its popularity
- React is not opinionated, which means that it won't force you to follow any specific design patterns, project organizational structure, or logic. It's all up to you.
- Smaller learning curve, especially when you already have a good grasp of JavaScript and HTML from our previous lessons.

> If you need more convincing, the recent [2022 StackOverflow Developer Survey](https://survey.stackoverflow.co/2022/#section-most-popular-technologies-web-frameworks-and-technologies) should be reason enough to get excited about learning React. Still unsure? Check out [this article](https://laptrinhx.com/top-10-reasons-why-you-should-learn-react-right-now-3718263995/).

Reviewer: Link text suggestion:
```suggestion
If you need more convincing, the recent [2022 StackOverflow Developer Survey](https://survey.stackoverflow.co/2022/#section-most-popular-technologies-web-frameworks-and-technologies) should be reason enough to get excited about learning React. Still unsure? Check out [this article on why you should learn React](https://laptrinhx.com/top-10-reasons-why-you-should-learn-react-right-now-3718263995/).
```

> _Convinced and excited? Great, because it's going to change your life! Let's dive right into it._

Reviewer: This sentence should be removed altogether since we won't be _diving_ into React until the next section.

```suggestion
```

<div class="lesson-content__panel" markdown="1">

> 1.  If you haven't already, browse through the [React Website](https://reactjs.org/). Don't go too indepth and dive into documentation however do read the introduction/homepage to gage an idea of how React works.

Reviewer: ```suggestion
1.  If you haven't already, browse through the [React Website](https://reactjs.org/). Don't go too indepth and dive into documentation however do read the introduction/homepage to gauge an idea of how React works.
```

1.  If you haven't already, browse through the <a href="https://reactjs.org/"> React Website </a>. Don't go too indepth and dive into documentation however do read the introduction/homepage to gage an idea of how React works.

2.  Read this <a href="https://www.freecodecamp.org/news/the-difference-between-a-framework-and-a-library-bd133054023f/"> FreeCodeCamp article </a> discussing the differences between a Javascript Library and a Framework

> 3.  Lastly, skim this <a href="https://www.peerbits.com/blog/reasons-to-choose-reactjs-for-your-web-development-project.html">in-depth article</a> which explains the major benefits of using React.

Reviewer: Can we have markdown formatted links here?

And I think this article could be a great assignment: https://blog.risingstack.com/the-history-of-react-js-on-a-timeline/

Author: Markdown formatted links have been added!

> 2.  Glance at [this article](https://blog.risingstack.com/the-history-of-react-js-on-a-timeline/) which outlines the history of React.
> 
> 3.  Read this [FreeCodeCamp article](https://www.freecodecamp.org/news/the-difference-between-a-framework-and-a-library-bd133054023f/) discussing the differences between a Javascript Library and a Framework
> 
> 4.  Lastly, skim this [in-depth article](https://www.peerbits.com/blog/reasons-to-choose-reactjs-for-your-web-development-project.html) which explains the major benefits of using React.

Reviewer: Link text improvements:

```suggestion
2.  Glance at [this article which outlines the history of React](https://blog.risingstack.com/the-history-of-react-js-on-a-timeline/).

3.  Read this [FreeCodeCamp article discussing the differences between a JavaScript library and framework](https://www.freecodecamp.org/news/the-difference-between-a-framework-and-a-library-bd133054023f/).

4.  Lastly, skim this [in-depth article which explains the major benefits of using React](https://www.peerbits.com/blog/reasons-to-choose-reactjs-for-your-web-development-project.html).
```

</div>

### Knowledge Check

This section contains questions for you to check your understanding of this lesson on your own. If you’re having trouble answering a question, click it and review the material it links to.

- <a class="knowledge-check-link" href="#what-is-react">What is the purpose of React?</a>

> - <a class="knowledge-check-link" href="#why-cover-react">What are two beenfits of using React?</a>

Reviewer: ```suggestion
- <a class="knowledge-check-link" href="#why-cover-react">What are benefits of using React?</a>
```
